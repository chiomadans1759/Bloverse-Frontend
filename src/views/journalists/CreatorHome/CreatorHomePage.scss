@import 'sass-rem';

#bl-creators{
	.top-section-container{
      color: #ffffff;
      min-height: rem(658px);
      background-image: radial-gradient(117% rem(900px) at 50% rem(-30px), #003CE1 76%, #2F80ED, transparent 75%);
		.welcome__link{
			h1.title{
				font-size: rem(38px);
				margin-top: rem(130px);
			}
			p{
				font-size: rem(16px);
    			margin-top: 1%;
			}
			a.btn-default{
				background: #ffffff;
				border-radius: rem(1px) !important;
    		padding: rem(7px) rem(30px);
			}

		}
	}
  .copyright {
    margin-top: rem(134px);
    .title {
      font-size: rem(14px);
      color: #B7B7B7;

      span {
        padding-left: rem(4px);
        font-size: rem(10px);
      }
    }
  }
	.metrics{
		.card{
			padding: 1.875rem;
		    border-radius: rem(130px) !important;
		    border-color: rgba(0, 0, 0, 0.15) !important;
		    margin-top: -12%;
			h1{
				font-size: rem(55px);
			}
			p{
				color: rgba(134, 134, 134, 1);
    		font-size: rem(14px);
			}
		}
	}
  .playVideo{
    margin-top: 7%;
  }
	.video-button {
      height: rem(80px);
      width: rem(80px);
      position: absolute;
      top: 50%;
      border: rem(1px) solid #2f80ed;
      background:#003ce1;
      left: 50%;
      animation-duration: 1s;
      animation-direction: alternate;
      animation-name: pop;
      animation-iteration-count: infinite;
      transform: translate(-50%, -50%);
      border-radius: 50%;
      
      .video-controls {
        position: absolute;
        top: 50%;
        left: 50%;
        transform: translate(-50%, -50%);
        color: #ffffff;
      }
      svg{
        color: #ffffff;
      }

      button {
        background-color: transparent;
        border: none;
        p{
          position: absolute;
          top: 102%;
          margin-top: 23%;
          font-weight: 400;
          margin-left: -8%;
          font-size: rem(15px);
          width: rem(100px);
        }
      }
    }
}

@keyframes pop{
  from{
    box-shadow: 0 0 0 rem(5px) #2f80ed;
  }
  to{
    box-shadow: 0 0 0 rem(15px) #2f80ed;
  }
}


@media( max-width : 585px ) {
#bl-creators{
  h1.title{
    font-size: rem(34px);
    margin-top: 10%;
    br{
      display: none !important;
    }
  }
  .metrics{
      .card{
        padding: 1.875rem;
        border-radius: rem(1px) !important;
        border-color: rgba(0, 0, 0, 0.15) !important;
        margin-top: rem(-196px) !important;
        margin-bottom: 19% !important;
        h1{
          font-size: rem(55px);
        }
        p{
          color: rgba(134, 134, 134, 1);
            font-size: rem(14px);
        }
      }
    }
  }
}

@media( max-width : 989px ) {
#bl-creators{
  h1.title{
    font-size: rem(34px);
    margin-top: 10%;
    br{
      display: none !important;
    }
  }
  .metrics{
      .card{
        padding: 1.875rem;
          border-radius: rem(1px) !important;
          border-color: rgba(0, 0, 0, 0.15) !important;
          margin-top: rem(-196px) !important;
          margin-bottom: 19% !important;
        h1{
          font-size: rem(55px);
        }
        p{
          color: rgba(134, 134, 134, 1);
            font-size: rem(14px);
        }
      }
    }
  }
}
