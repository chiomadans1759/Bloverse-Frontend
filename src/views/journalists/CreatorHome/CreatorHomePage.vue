<template>
  <main id="bl-creators" class="container-fluid p-0">
    <!-- Top Section Container Start -->
    <div class="top-section-container">
      <Navbar isTransparent whiteText/>
      <div class="container">
        <div class="welcome__link">
          <h1 class="title">Interactive stories from
            <br>the global community
          </h1>
          <p class="paragraph">Connecting creators to the people</p>
          <br>
          <div class="publish-link-container mb-3 mb-sm-0">
            <router-link to="/creators/apply" class="btn btn-default">Apply</router-link>
          </div>
        </div>
      </div>
    </div>
    <div class="metrics">
      <div class="container">
        <div class="row">
          <div class="col-md-2"></div>
          <div class="col-md-8">
            <div class="card">
              <div class="row text-center">
                <div class="col-md-4 col-12 col-sm-4 col-lg-4">
                  <h1>5.6m</h1>
                  <p>Content Published</p>
                </div>
                <div class="col-md-4 col-12 col-sm-4 col-lg-4">
                  <h1>5.6k</h1>
                  <p>Content Creator</p>
                </div>
                <div class="col-md-4 col-12 col-sm-4 col-lg-4">
                  <h1>1.8m</h1>
                  <p>Visitors</p>
                </div>
              </div>
            </div>
          </div>
          <div class="col-md-2"></div>
        </div>
      </div>
    </div>
    <div class="playVideo">
      <div class="container">
        <div class="row">
          <div class="col-md-3"></div>
          <div class="col-md-6">
            <video class="col-12" id="video-element" controls style="display: none;">
              <source src="@/assets/photocontest.mp4" type="video/mp4">
            </video>
            <div v-if="paused && !playing" class="video-button play">
              <button v-if="paused && !playing" @click="play">
                <i class="fas fa-play video-controls"></i>
                <p>Watch Video</p>
              </button>
            </div>
          </div>
          <div class="col-md-3"></div>
        </div>
      </div>
    </div>
    <div class="copyright text-center mt-13">
      <p class="title">
        <b>bloverse</b>
        <span class="sub-text">2018. All Right Reserved</span>
      </p>
    </div>
      <modal
        target="feedback-modal"
        altTitle="Send Us Some Feedback"
        actionText="Send Feedback"
        dark
        subTitle="Found a bug? Have a suggestion?. Fill out the form below and we'll take a look!"
        >
        <div class="bl-modal-content">
          <p>Enter your feedback here!</p>
          <textarea class="form-control mb-2 p-1"></textarea>
          <label class="ml-1 mr-3"><input type='radio' name='feedback-type'>  Bug</label>
          <label class="mr-3"><input type='radio' name='feedback-type'>  Comment</label>
          <label class="mr-3"><input type='radio' name='feedback-type'>  Other</label>
        </div>
      </modal>
  </main>
</template>

<script>
import Navbar from "@/components/Navbar/Navbar.vue";
import Modal from "@/components/Modal/Modal";

export default {
  name: "CreatorHomePage",
  data() {
    return {
      playing: false,
      paused: true
    };
  },
  components: { 
    Navbar,
    Modal,
  },
  methods: {
    play: function() {
      const video = document.getElementById("video-element");
      video.style.display = "block";
      this.playing = true;
      this.paused = false;
      video.play();
    }
  },
  mounted() {
    const video = document.getElementById("video-element");
    const self = this;
    if (video) {
      video.onpause = function() {
        self.playing = false;
        self.paused = true;
      };
      video.onplay = function() {
        self.playing = true;
        self.paused = false;
      };
    }
  }
};
</script>

<style lang="scss" scoped>
@import "./CreatorHomePage.scss";
</style>
