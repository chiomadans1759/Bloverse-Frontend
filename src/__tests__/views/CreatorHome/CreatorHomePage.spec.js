import { mount, RouterLinkStub  } from '@vue/test-utils'
import CreatorHomePage from '@/views/journalists/CreatorHome/CreatorHomePage.vue';

describe('Creator Home Page', () => {
  let wrapper = mount(CreatorHomePage, {
    stubs: {
      RouterLink: RouterLinkStub
    },
  });

  it('should render correctly', () => {
    expect(wrapper.find(RouterLinkStub).props().to).toBe('/')
    expect(wrapper.element).toMatchSnapshot();
  });

  describe('Computed', () => {
    it('should get all images', (done) => {

      expect(wrapper.vm.getImages).toHaveLength(20);
      done();
    });

    it('should set the top section style', () => {

      expect(wrapper.vm.topSectionStyle.opacity).toBe(1);
      wrapper.setData({playing: true});
      expect(wrapper.vm.topSectionStyle.opacity).toBe(0.4);
    });
  });

  it('should play the video', () => {
    const playStub = jest.spyOn(window.HTMLMediaElement.prototype, 'play').mockImplementation(() => {});
    wrapper.setData({playing: false});
    wrapper.find('.video-button').trigger('click');

    expect(wrapper.vm.playing).toBeTruthy();
    expect(wrapper.vm.paused).toBeFalsy();
    expect(playStub).toHaveBeenCalled();
    playStub.mockRestore();
  });

  it('should close the video', () => {
    const pauseStub = jest.spyOn(window.HTMLMediaElement.prototype, 'pause').mockImplementation(() => {});
    wrapper.find('.creator-video').trigger('blur');

    expect(wrapper.vm.playing).toBeFalsy();
    expect(wrapper.vm.paused).toBeTruthy();
    expect(pauseStub).toHaveBeenCalled();
    pauseStub.mockRestore();
  });

});
