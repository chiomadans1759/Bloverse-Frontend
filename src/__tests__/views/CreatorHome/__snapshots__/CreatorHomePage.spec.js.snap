// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Creator Home Page should render correctly 1`] = `
<main
  class="container-fluid p-0"
  id="bl-creators"
>
  <div
    class="top-section-container"
  >
    <header
      id="feeds-header"
    >
      <nav
        class="navbar navbar-expand-lg navbar-light px-5 transparent alt-button"
      >
        <div
          class="container-fluid"
        >
          <!---->
           
          <span
            class="back-text"
          >
            Back
          </span>
           
          <a
            class="router-link"
          >
            <!---->
             
            <img
              class="nav-logo"
              src="@/assets/Bloverse-white.svg"
            />
          </a>
           
          <div
            aria-controls="navbarSupportedContent"
            aria-expanded="false"
            aria-label="Toggle navigation"
            class="form-inline my-2 my-lg-0 navbar-toggler mobile-view"
            data-target="#navbarSupportedContent"
            data-toggle="collapse"
          >
            <a
              aria-expanded="false"
              aria-haspopup="true"
              class="nav-link dropdown-toggle"
              data-toggle="dropdown"
              href="#"
              id="navbarDropdown"
              role="button"
            >
              <span
                id="option-icon"
              />
               
              <span
                id="option-icon"
              />
               
              <span
                id="option-icon"
              />
            </a>
             
            <div
              aria-labelledby="navbarDropdown"
              class="dropdown-menu dropdown-mobile-view"
            >
              <a
                class="dropdown-item"
              >
                About
              </a>
               
              <a
                class="dropdown-item"
              >
                Photo Contest
              </a>
               
              <a
                class="dropdown-item"
              >
                Terms
              </a>
               
              <a
                class="dropdown-item"
              >
                Privacy
              </a>
               
              <a
                class="dropdown-item"
                href="#"
              >
                FAQ
              </a>
               
              <div
                class="dropdown-divider"
              />
               
              <a
                class="dropdown-item"
                data-toggle="modal"
                href="#feedback-modal"
              >
                Feedback
              </a>
            </div>
             
            <div>
              <span
                class="vl bg-light"
              />
               
              <!---->
            </div>
             
            <div
              class="nav-item dropdown initials"
            >
              <a
                aria-expanded="false"
                aria-haspopup="true"
                class="dropdown-toggle initials-link round-avatar"
                data-toggle="dropdown"
                href="#"
                id="navbarDropdown"
                role="button"
                to="/login"
              >
                MD
              </a>
               
              <div
                aria-labelledby="navbarDropdown"
                class="dropdown-menu dropdown-mobile-view__avatar"
              >
                <a
                  class="dropdown-item"
                >
                  <span
                    class="icon-profile"
                  >
                    <i
                      class="fal fa-user-circle"
                    />
                  </span>
                  
            Profile
          
                </a>
                 
                <div
                  class="dropdown-divider"
                />
                 
                <a
                  class="dropdown-item"
                >
                  <span
                    class="icon-settings"
                  >
                    <i
                      class="fal fa-cog"
                    />
                  </span>
                  
            Settings
          
                </a>
                 
                <div
                  class="dropdown-divider"
                />
                 
                <a
                  class="dropdown-item"
                  href="#"
                >
                  <span
                    class="icon-power"
                  >
                    <i
                      class="far fa-power-off"
                    />
                  </span>
                  
            Sign Out
          
                </a>
              </div>
            </div>
          </div>
           
          <div
            class="collapse navbar-collapse"
          >
            <ul
              class="navbar-nav ml-auto white-text"
            >
              <li
                class="nav-item active"
              >
                <a
                  class="nav-link"
                >
                  Home
            
                  <span
                    class="sr-only"
                  >
                    (current)
                  </span>
                </a>
              </li>
               
              <li
                class="nav-item"
              >
                <a
                  class="nav-link"
                >
                  About
                </a>
              </li>
               
              <li
                class="nav-item"
              >
                <a
                  class="nav-link"
                  href="/photocontest"
                >
                  Photo Contest
                </a>
              </li>
               
              <li
                class="nav-item dropdown pr-4"
              >
                <a
                  aria-expanded="false"
                  aria-haspopup="true"
                  class="nav-link dropdown-toggle"
                  data-toggle="dropdown"
                  href="#"
                  id="navbarDropdown"
                  role="button"
                >
                  <span
                    id="option-icon"
                  />
                   
                  <span
                    id="option-icon"
                  />
                   
                  <span
                    id="option-icon"
                  />
                </a>
                 
                <div
                  aria-labelledby="navbarDropdown"
                  class="dropdown-menu dropdown-menu--elipsis"
                >
                  <a
                    class="dropdown-item"
                  >
                    Terms
                  </a>
                   
                  <a
                    class="dropdown-item"
                  >
                    Privacy
                  </a>
                   
                  <a
                    class="dropdown-item"
                    href="#"
                  >
                    FAQ
                  </a>
                   
                  <div
                    class="dropdown-divider"
                  />
                   
                  <a
                    class="dropdown-item"
                    data-toggle="modal"
                    href="#feedback-modal"
                  >
                    Feedback
                  </a>
                </div>
              </li>
            </ul>
             
            <div
              class="form-inline my-2 my-lg-0 pr-4 alt-button"
            >
              <!---->
               
              <div
                class="nav-item dropdown initials"
              >
                <a
                  aria-expanded="false"
                  aria-haspopup="true"
                  class="dropdown-toggle d-flex justify-content-center align-items-center"
                  data-toggle="dropdown"
                  href="#"
                  id="navbarDropdown"
                  role="button"
                  to="/login"
                >
                  <button
                    class="round-avatar mr-2"
                  >
                    MD
                  </button>
                   
                  <i
                    class="fal fa-angle-down fa-2x down-arrow"
                  />
                </a>
                 
                <div
                  aria-labelledby="navbarDropdown"
                  class="dropdown-menu"
                >
                  <a
                    class="dropdown-item"
                  >
                    <span
                      class="icon-profile"
                    >
                      <i
                        class="fal fa-user-circle"
                      />
                    </span>
                    
              Profile
            
                  </a>
                   
                  <div
                    class="dropdown-divider"
                  />
                   
                  <a
                    class="dropdown-item"
                  >
                    <span
                      class="icon-settings"
                    >
                      <i
                        class="fal fa-cog"
                      />
                    </span>
                    
              Settings
            
                  </a>
                   
                  <div
                    class="dropdown-divider"
                  />
                   
                  <a
                    class="dropdown-item"
                    href="#"
                  >
                    <span
                      class="icon-power"
                    >
                      <i
                        class="far fa-power-off"
                      />
                    </span>
                    
              Sign Out
            
                  </a>
                </div>
              </div>
               
              <span
                class="vl bg-light"
              />
               
              <a
                class="publish-link btn btn-sm"
              >
                Publish
              </a>
            </div>
          </div>
        </div>
      </nav>
       
      <main
        id="bl-modal"
      >
        <div
          aria-hidden="true"
          aria-labelledby="customModal"
          class="modal fade"
          id="feedback-modal"
          role="dialog"
          tabindex="-1"
        >
          <div
            class="modal-dialog modal-dialog-centered"
            role="document"
          >
            <div
              class="modal-content"
            >
              <div
                class="modal-header darkBackground"
              >
                <!---->
                 
                <div>
                  <h6>
                    <strong
                      class="alt-title"
                    >
                      Send Us Some Feedback
                    </strong>
                  </h6>
                   
                  <h6
                    class="mt-2 sub-title"
                  >
                    Found a bug? Have a suggestion?. Fill out the form below and we'll take a look!
                  </h6>
                </div>
                 
                <button
                  aria-label="Close"
                  class="close"
                  data-dismiss="modal"
                  type="button"
                >
                  <span
                    aria-hidden="true"
                  >
                    ×
                  </span>
                </button>
              </div>
               
              <div
                class="modal-body"
              >
                <div
                  class="bl-modal-content"
                >
                  <p>
                    Enter your feedback here!
                  </p>
                   
                  <textarea
                    class="form-control mb-2 p-1"
                  />
                   
                  <label
                    class="ml-1 mr-3"
                  >
                    <input
                      name="feedback-type"
                      type="radio"
                    />
                      Bug
                  </label>
                   
                  <label
                    class="mr-3"
                  >
                    <input
                      name="feedback-type"
                      type="radio"
                    />
                      Comment
                  </label>
                   
                  <label
                    class="mr-3"
                  >
                    <input
                      name="feedback-type"
                      type="radio"
                    />
                      Other
                  </label>
                </div>
              </div>
               
              <div
                class="modal-footer"
              >
                <button
                  class="btn btn-outline-primary"
                  data-dismiss="modal"
                  type="button"
                >
                  
              Close
            
                </button>
                 
                <button
                  class="btn btn-primary"
                  type="button"
                >
                  Send Feedback
                </button>
              </div>
            </div>
          </div>
        </div>
      </main>
    </header>
     
    <div
      class="container"
    >
      <div
        class="welcome__link"
      >
        <h1
          class="title"
        >
          Interactive stories from
          
          <br />
          the global community
        
        </h1>
         
        <p
          class="paragraph"
        >
          Connecting creators to the people
        </p>
         
        <br />
         
        <div
          class="publish-link-container mb-3 mb-sm-0"
        >
          <a
            class="btn btn-default"
          >
            Apply
          </a>
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="metrics"
  >
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col-md-2"
        />
         
        <div
          class="col-md-8"
        >
          <div
            class="card"
          >
            <div
              class="row text-center"
            >
              <div
                class="col-md-4 col-12 col-sm-4 col-lg-4"
              >
                <h1>
                  5.6m
                </h1>
                 
                <p>
                  Content Published
                </p>
              </div>
               
              <div
                class="col-md-4 col-12 col-sm-4 col-lg-4"
              >
                <h1>
                  5.6k
                </h1>
                 
                <p>
                  Content Creator
                </p>
              </div>
               
              <div
                class="col-md-4 col-12 col-sm-4 col-lg-4"
              >
                <h1>
                  1.8m
                </h1>
                 
                <p>
                  Visitors
                </p>
              </div>
            </div>
          </div>
        </div>
         
        <div
          class="col-md-2"
        />
      </div>
    </div>
  </div>
   
  <div
    class="playVideo"
  >
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col-md-3"
        />
         
        <div
          class="col-md-6"
        >
          <video
            class="col-12"
            controls="controls"
            id="video-element"
            style="display: none;"
          >
            <source
              src="@/assets/photocontest.mp4"
              type="video/mp4"
            />
          </video>
           
          <div
            class="video-button play"
          >
            <button>
              <i
                class="fas fa-play video-controls"
              />
               
              <p>
                Watch Video
              </p>
            </button>
          </div>
        </div>
         
        <div
          class="col-md-3"
        />
      </div>
    </div>
  </div>
   
  <div
    class="copyright text-center mt-13"
  >
    <p
      class="title"
    >
      <b>
        bloverse
      </b>
       
      <span
        class="sub-text"
      >
        2018. All Right Reserved
      </span>
    </p>
  </div>
</main>
`;
